File: final_project/backend/db/crud/shopping_cart.go
package crud

import (
	"fmt"
	"backend/db"
)

// Create Shopping Cart
func CreateShoppingCart(userID uint) {
	cart := db.ShoppingCart{UserID: userID}
	result := db.DB.Create(&cart)
	if result.Error != nil {
		fmt.Println("Error creating shopping cart:", result.Error)
		return
	}
	fmt.Println("Shopping cart created successfully:", cart)
}

// Get All Shopping Carts
func GetAllShoppingCarts() {
	var carts []db.ShoppingCart
	result := db.DB.Find(&carts)
	if result.Error != nil {
		fmt.Println("Error fetching shopping carts:", result.Error)
		return
	}
	fmt.Println("Shopping carts:", carts)
}

// Delete Shopping Cart
func DeleteShoppingCart(cartID uint) {
	result := db.DB.Delete(&db.ShoppingCart{}, cartID)
	if result.Error != nil {
		fmt.Println("Error deleting shopping cart:", result.Error)
		return
	}
	fmt.Println("Shopping cart deleted successfully.")
}

File: final_project/frontend/src/services/shoppingCart.js
import api from "./api";

// ShoppingCart Services
export const fetchShoppingCarts = async () =>
  api.get("/shopping-carts").then((res) => res.data);

export const fetchShoppingCartById = async (id) =>
  api.get(`/shopping-carts/${id}`).then((res) => res.data);

export const createShoppingCart = async (cartData) =>
  api.post("/shopping-carts", cartData).then((res) => res.data);

export const deleteShoppingCart = async (id) =>
  api.delete(`/shopping-carts/${id}`).then((res) => res.status === 204);
